### PR TITLE
Rollback to pse halo2 and halo2wrong for first release

### LINKED ISSUE #2: Release with commitment scheme abstraction and challenge API
Depends on https://github.com/privacy-scaling-explorations/halo2/issues/91 and https://github.com/privacy-scaling-explorations/halo2/issues/90

### LINKED ISSUE #6: Load commitments of preprocessed polynomial outside accumulation
Currently in halo2 loader the commitments of preprocessed polynomial are baked into circuit as constants, but to implement PCD (atomic accumulation) instead of just aggregation, they should be loaded as witness to make sure the preprocessed part doesn't depend on itself.

To achieve this we can make halo2 loader to cache loaded commitments of preprocessed polynomial (as witness).

### LINKED ISSUE #8: Remove cargo patch sections and depend on the forks explicitly
Patch sections are ignored if this crate is added as a dependency, hence remove the patch sections and depend on any forks without overriding.

### LINKED ISSUE #9: Re-export dependencies
The following packages are needed and it's useful if the `plonk-verifier` package exports (the correct) versions of them:

- halo2_wrong_ecc
- halo2_wrong_transcript

### LINKED ISSUE #11: Invalid elliptic curve point encoding in proof
This happens with [this test](https://github.com/privacy-scaling-explorations/zkevm-chain/blob/3b933bcd4df22e4ec5833315a7a1c17953186c09/prover/tests/verifier.rs#L130)

```
---- autogen_aggregation_verifier stdout ----
thread 'autogen_aggregation_verifier' panicked at 'called `Result::unwrap()` on an `Err` value: Transcript(Other, "Invalid elliptic curve point encoding in proof")', /app/prover/src/aggregation_circuit.rs:213:26
```

### PR DESCRIPTION
**Pull request by han0110**
Rollback without challenge API for the first release.

Resolves #2. Resolves #6. Resolves #8. Resolves #9. Resolves #11.

### Todo

- [x] Merge https://github.com/privacy-scaling-explorations/halo2/pull/96
- [x] Merge https://github.com/privacy-scaling-explorations/halo2wrong/pull/36
- [x] Merge https://github.com/privacy-scaling-explorations/halo2wrong/pull/38

### CODE REVIEW

File: src/pcs/kzg/accumulation.rs [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        pk: &Self::ProvingKey,
+        instances: &[PCS::Accumulator],
+        transcript: &mut T,
+        rng: R,
+    ) -> Result<PCS::Accumulator, Error>
+    where
+        T: TranscriptWrite<C>,
+        R: Rng,
+    {
+        assert!(instances.len() > 1);
```
Comment (pinkiebell) [inline src/pcs/kzg/accumulation.rs:156]: @han0110 I'm hitting this one. Shouldn't that be `instances.len > 0`?

**Comment by pinkiebell** [comment]
@han0110 Maybe you can cherry-pick [this one](https://github.com/pinkiebell/plonk-verifier/commit/8a2e7fb01f6975036b5122893aa8829f93b5c111) 🙂

File: src/pcs/kzg/accumulation.rs (hunk 1 above)
Comment (han0110) [reply src/pcs/kzg/accumulation.rs]: Make sense, updated in https://github.com/privacy-scaling-explorations/plonk-verifier/pull/5/commits/5a5a6478525817aaaf05b305871cfca7b0704261.

**Comment by han0110** [comment]
> @han0110 Maybe you can cherry-pick [this one](https://github.com/pinkiebell/plonk-verifier/commit/8a2e7fb01f6975036b5122893aa8829f93b5c111) 🙂

I have the aux generator automatically assigned when doing the msm in https://github.com/privacy-scaling-explorations/plonk-verifier/pull/5/commits/b1fa31812948d49cf13a76af2c9213538ae833ce, I think the api is even simpler, so I didn't pick this one in the end.

**Pull request merged**